### DISCUSSION TITLE
Can Ldown be forced from insitu measurements when using SOLWEIG ?

### DISCUSSION THREAD

**Discussion by ML426** [UMEP-dev/UMEP#587] [Q&A]
Hi,
Is it possible to force in my simulation the mesured Ldown from insitu measurements rather than using the prata approximation or something else?
I've tried forcing it by modifying the Ldown column in my weather file but I haven't seen much difference in my Ldown results (between when I leave -999 and when I use my measured values).

Furthermore, in the emissivity model, would it be possible to directly test another formula, for example Angston's formula, with different coefficients a, b and c for day and night? 

Regards,
ML

PS: I am indeed referring to Ldown originating from the sky before interactions with the urban scene where it may be influenced by surface reflections. By the way, could you confirm for me that in the calculation of Ldown on the horizontal plane of the scene, it is indeed a component integrating LW from the sky and the surrounding surfaces? I think I saw something similar in the code, but as I write, I have a doubt :)

**Comment by biglimp** [comment]
Hi,
At the moment, we only have one method to estimate Ldown (Prata in conjunction with Crawford and Duchon [[Lindberg et al. 2008](https://link.springer.com/article/10.1007/s00484-008-0162-7)]). It would be somewhat straight forward to include other methods but you need to manipulate the code and the gui. @nilswallenberg, is this something we should do for the upcoming version or @ML426 , are you familiar with python and want to contribute?

The model output L component are estimated 1.1 magl so it includes surrounding surfaces (if there are any).

**Comment by nilswallenberg** [comment]
Hi @ML426,

Do you have a reference for Angston (Ångström?)?

Cheers,
Nils

**Reply by ML426** [reply]
Hi @nilswallenberg ,
Yes, Ångström

**Comment by ML426** [comment]
Hi @biglimp,

Yes, I am familiar with Python (not specifically Python on QGIS), but I am able to understand the code in a general sense.

I have a measurement campaign that I am comparing to Solweig, and I'm noticing significant discrepancies especially at night. So, I'm questioning the possibility of improving the model. During the day, the anisotropic model works well, but at night, the isotropic model seems to perform better. In the anisotropic model, Ldown is substantially improved during the day but degraded at night. In Nil's paper, he mentions the possibility that emissivity may have influenced the results. I would like to delve into this by testing with the measurement data from my experimental campaign and comparing it with other models. I've seen in the code that there are two other emissivity models. Therefore, I want to see if testing other models or forcing measured Ldown could lead to better results (while waiting for advancements in estimating Ts).

Additionally, I admit that diving into this extensive codebase can be time-consuming. If I have support and guidance, yes, I am willing to test and contribute to improvements.

**Reply by biglimp** [reply]
Interesting. I think @nilswallenberg is already on it. HE says he will include obs and ångström next week. Where are your observations from? I you like we can also share some of our observations for evaluation.

**Reply by ML426** [reply]
In Paris, on a canyon street

**Reply by biglimp** [reply]
Your bad performance at night is probably due to bad modelling of nighttime wall temperatures. We are about to update the model with a new surface temperature scheme that could be a solution. In the meantime, we should test your canyon with observed Ldown.

**Reply by ML426** [reply]
A few days ago, it was possible to obtain the hemispheric distribution at a point in the model output, which was very useful for understanding the results, but when I ran the simulation again yesterday, I no longer obtained this information. Has there been a change in the code? If so, is it possible to restore this option? Also, if it were possible to have L and K in the same repartition, it would be a great help in understanding the results.

![POI_1 5](https://github.com/UMEP-dev/UMEP/assets/87750674/4f29bb58-eed5-4895-bbcc-451c8dbecb89)
![metCheck](https://github.com/UMEP-dev/UMEP/assets/87750674/bd14c601-4e39-4042-8452-463841bf2dbd)

**Comment by ML426** [comment]
I'm all in for testing together :)

> Your bad performance at night is probably due to bad modelling of nighttime wall temperatures. We are about to update the model with a new surface temperature scheme that could be a solution. In the meantime, we should test your canyon with observed Ldown.

I'm all in for testing together :)

**Reply by biglimp** [reply]
The polarplot should be there but only if you use POI.

**Reply by ML426** [reply]
That was the case, but I had more at the output. 
Maybe it's a mistake on my part. I'll try using UMEP for processing.

Thanks

**Comment by biglimp** [comment]
We are a bit concerned with your radiation data it looks like you have tow values at each timestep?

**Reply by ML426** [reply]
The radiation data is recorded every minute.
We have one value for incident radiation (shortwave and longwave) on the roof.
And measurements in the canyon with 3 net radiometers, up and down (SW and LW) for 6 directions method.

**Reply by biglimp** [reply]
Ok. I is just that the plot you show looks strange. Can you show a plot of the input data Kdown

**Comment by ML426** [comment]
No, on the previous graph, it's an output from Solweig that compares Kglobale with I0.
Here's the radiation graph from the sensor placed on the roof. It's the only radiation sensor that measures every 15 minutes. All other radiation sensors measure every minute.

![image](https://github.com/UMEP-dev/UMEP/assets/87750674/89b79fde-e6fa-48af-8304-86fb418784fa)

**Reply by nilswallenberg** [reply]
Can up attach your meteorological input file for SOLWEIG here, please.

**Reply by ML426** [reply]
Ci-joint, mon fichier météo
[Meteo_file_UMEP_SANSDIFFUS_30min_3au8Sept.txt](https://github.com/UMEP-dev/UMEP/files/14222612/Meteo_file_UMEP_WithoutDIFFUS_30min_3au8Sept.txt)

**Reply by nilswallenberg** [reply]
Hmm, can't see anything. Only says "Not Found".

**Reply by ML426** [reply]
[Meteo_file_UMEP_SANSDIFFUS_30min_3au8Sept.txt](https://github.com/UMEP-dev/UMEP/files/14222715/Meteo_file_UMEP_SANSDIFFUS_30min_3au8Sept.txt)

**Reply by ML426** [reply]
Can also try to send you an email on your adress xxx.gvc.gu.se

**Reply by nilswallenberg** [reply]
I see it now :)

**Reply by nilswallenberg** [reply]
Looks good! Have to change my code for the plot. :)

**Comment by nilswallenberg** [comment]
Okay, so next week I will add the Ångström equation for emissivity and you can try it. I will try to add an option to use observed Ldown, but I have to figure out how to handle the anisotropic sky with observed values. :)

**Reply by ML426** [reply]
Excellent, thank you Nils. 
Once you had managed to deal with the observed, I will test and share the results as soon as possible. :)

**Comment by nilswallenberg** [comment]
Hi @ML426,

Can you give me a specific reference for the Ångström equation you want to test?

Cheers,
Nils

**Reply by ML426** [reply]
Ldown =  Cte_Boltzman* Temp^4 * (a - b * 10^(-c*waterVapParPres))

with:

Cte_Boltzman= (5.67 × 10^-8) 
day: a = 0.9755, b = 0.4144, c = 0.04467
night: a = 0.8496, b = 0.9935, c = 0.1862

From Ångström, 1915.

**Comment by nilswallenberg** [comment]
Hi @ML426,

Here's a new version. I added the possibility to run with observed Ldown as well as sky longwave estimations according to Ångström (two from 1915?), Idso (1981) and Idso & Jackson (1969). I added dois for the papers in the code (emissivity_models.py). If you have another reference for the Ångström equation you proposed above (day and night) it would be nice, or if you could point to what page in the paper/thesis. Couldn't really find it in his paper/thesis.

Please, try it and see if it works as expected. I did some quick tests and I think it works, but who knows. :) Looking forward to hearing from you! 
[UMEP-processing-main.zip](https://github.com/UMEP-dev/UMEP/files/14277665/UMEP-processing-main.zip)

Cheers,
Nils

**Comment by ML426** [comment]
Hi @nilswallenberg,

Sorry for my late reply. I was a bit busy on other subjects.
This morning, I just tried to install the zip file but I got an error message on Qgis.
Last time when I had a simular problem, I updated the Qgis version to correct it. I had version 3.32.3 and I've just upgraded to version 3.34.3 but still nothing.

**Comment by ML426** [comment]
![image](https://github.com/UMEP-dev/UMEP/assets/87750674/9e88f0a5-5a3d-479d-a9f6-44864fe77984)

The error message

**Reply by nilswallenberg** [reply]
Ah, okay. I accidentally added one  directory too many. Try this one.
[UMEP-processing-main.zip](https://github.com/UMEP-dev/UMEP/files/14343481/UMEP-processing-main.zip)

**Comment by ML426** [comment]
@nilswallenberg

The installation was successful, thank you :)

Unfortunately, I've just tested it and got an error message :

```
Traceback (most recent call last):
File "C:\Users/xxxxxx/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\UMEP-processing-main\processor\solweig_algorithm.py", line 892, in processAlgorithm
diffsh[:, :, i] = shmat[:, :, i] - (1 - vegshmat[:, :, i]) * (1 - transVeg) # changes in psi not implemented yet
ValueError: could not broadcast input array from shape (89,145) into shape (424,477)
```
 
I think it comes from the part where the diffuse radiation for the anisotropic model is calculated taking into account the vegetation DEM. I have a cDSM and a tDSM in the study case I want to test.

**Reply by nilswallenberg** [reply]
I see. Maybe you have to recalculate SVF if you used an older version of UMEP previously.

**Comment by ML426** [comment]
Exactly, I had kept the results of the previous SVF. But the calculations were already with the anisotropic model and a subdivision with 153 patches. Has anything changed in the SVFs calculation?

In the meantime, I'll restart the SVF calculations and I'll let you know.
Thx :)

**Comment by ML426** [comment]
Hi @nilswallenberg,
Thanks, i was able to run the version you sent me. It works very well.

The models for emissivity show a slight difference in the total Ldown on the street (which also takes into account the influence of buildings). I am preparing a documented report to share with you, and I am also taking the opportunity to test other case studies.

To compare with my pyrgeometer on the roof, do I need to adjust the height of the figure to the height of my roof (z=6m instead of 1.1m), or is it enough to place my Point Of Interest at the location of the wall?

Another very important question, I notice differences between the outputs from UMEP for processing and the already installed UMEP. Is this normal? Has anything else changed in the calculation of radiation, or SVFs ? When comparing the hemispheric distribution outputs, where patches are categorized into distinct classes such as sky, vegetation, shaded building walls, etc., subtle differences are observed despite using identical inputs (DSM, cDSM, SVF...).

**Reply by nilswallenberg** [reply]
Good to hear! Looking forward to seeing some results. :)

To compare with your pyrgeometer on the roof you only compare the Ldown output from SOLWEIG on the roof. What I did now is that I use the input Ldown and divided (for the sky) to the patches (anisotropic). 

Hmm, not sure what causes these differences. Could be because of different versions of UMEP (SOLWEIG/SVF calculator). Best is to use all tools within one version of UMEP. For example, we had 145 patches in previous versions of SOLWEIG, whereas we now have 153 patches. If you run SOLWEIG for a version using 153 patches, but with SVF calculated with only 145 you will get a divergence.

Cheers,
Nils

**Comment by ML426** [comment]
If I position a receiver (POI) at the same location as the wall, which in this case is my roof, the Ldown value recorded in the output file aligns with what I would expect from a pyrgeometer. That is right? 
I had understood that all Li and Ki values were meant to be at a height of 1.1 meters. Specifically, on my roof, is my SVF set to 1 for the entire patch to ensure that no obstacles are visible from this location?

As for the discrepancies I've noticed, I'm utilizing the most recent version of UMEP, or at least the version that is current when using the plugin. I had previously executed my models using the anisotropic sky and the default Prata model. I've rerun the model under the same conditions but this time with the UMEP for processing version you've recently sent me (maintaining the same inputs). Nevertheless, I observed different outcomes between the two attempts. My goal was expressly to draw comparisons with my earlier results.
Upon comparison, I noticed that the hemispherical outputs weren't exactly identical, which led to my inquiry.


![POI_1](https://github.com/UMEP-dev/UMEP/assets/87750674/f15fd00a-1767-4e67-bcb9-145aee1d11e2)
![POI_1](https://github.com/UMEP-dev/UMEP/assets/87750674/1be0ebe1-e894-4685-addd-92e83d5cd894)

Best,
ML

**Reply by nilswallenberg** [reply]
Ah, yes. This might have to do with a bug concerning the position of the POI (it was off by a pixel). It is fixed in the version available on GitHub.

**Comment by ML426** [comment]
Alright, I see.
However, in my case, this leads to points that end up in the shadow or in the sunlight when, in practice, this is not the case.
Just to allow me to compare under the same conditions the impact of the sky model on emissivity and to assess the error induced by one method or the other, as well as the impact of the corrections made, is it possible to get a version identical to the one currently available on QGIS?
Otherwise, could you please point out the lines of code that have changed?
